test/orthonormal_bases.jl

```diff
@@ -94,3 +94,92 @@ struct NonBasis <: Manifolds.AbstractBasis{ℝ} end
         @test_throws ArgumentError get_basis(A, pts[1], PrecomputedOrthonormalBasis([2*pts[1], pts[1], pts[1]]))
     end
 end
+
+@testset "Basis show methods" begin
+    @test sprint(show, ArbitraryOrthonormalBasis()) == "ArbitraryOrthonormalBasis(ℝ)"
+    @test sprint(show, ArbitraryOrthonormalBasis(ℂ)) == "ArbitraryOrthonormalBasis(ℂ)"
+    @test sprint(show, ProjectedOrthonormalBasis(:svd)) == "ProjectedOrthonormalBasis(:svd, ℝ)"
+    @test sprint(show, ProjectedOrthonormalBasis(:gram_schmidt, ℂ)) == "ProjectedOrthonormalBasis(:gram_schmidt, ℂ)"
+
+    @test sprint(show, "text/plain", DiagonalizingOrthonormalBasis(Float64[1, 2, 3])) == """
+    DiagonalizingOrthonormalBasis with coordinates in ℝ and 0 curvature in direction:
```

Comment:
Zero curvature in direction is a little imprecise, but maybe the long explanation is too much for a `show`:
The DiagonalizingONB is an ONB in one tangent space equipped with eigenvalues of the curvature tensor that characterise how much a tangent vector changes when being transported along the geodesic given by the first tangent vector. That way the first tangent vector gets a zero, since its the covariant derivtative of the geodesic considered anyway.

The imprecision is now, that that does not mean zero curvature, the manifold is still (happily) curved as before, just the curvature tensor when restricted to this orthonormal Frame (=set of ONBs along the geodesic by parallel transport) has an eigenvalue of zero for that direction. And yes this is far too long for a print, so maybe just FYI.

Reply:
Thanks, that's helpful. I chose this wording as a synopsis of what the docs say, that is, it's the basis "...that diagonalizes the curvature tensor $R(u,v)w$ and where the direction `v` has curvature `0`.`"

I'm happy to update both the docs and the `show` method, with the latter remaining not much longer than it currently is. What would you propose?

Reply:
the shortest way would be to replace curvature by eigenvalue?

Reply:
Okay, I made a fix; let me know if it resolves this.